tsl/test/sql/dist_function_pushdown.sql

```diff
@@ -0,0 +1,124 @@
+\c :TEST_DBNAME :ROLE_CLUSTER_SUPERUSER;
+\o
+SET client_min_messages TO NOTICE;
+\set ECHO all
+
+
+-- Test pushdown of stable functions to data nodes
+create table words(ts timestamp, text text);
+select create_distributed_hypertable('words', 'ts');
+insert into words values ('2021-01-01 01:01:01', 'testing'),
+	('2022-02-02 02:02:02', 'some other words');
+	
+-- The normal stuff, to_tsvector is evaluated on the access node. We'll check
+-- different text search configurations and make sure that they give different
+-- results.
+set timescaledb.pushdown_functions = 'whitelisted';
+set default_text_search_config = 'english';
+
+select to_tsvector(text), to_tsvector('testing')
+from words where to_tsvector(text) != to_tsvector('testing');
+
+explain (analyze, verbose, costs off, timing off, summary off)
+select to_tsvector(text), to_tsvector('testing')
+from words where to_tsvector(text) != to_tsvector('testing');
+
+-- A different stemming configuration.
+set default_text_search_config = 'simple';
+
+select to_tsvector(text), to_tsvector('testing')
+from words where to_tsvector(text) != to_tsvector('testing');
+
+explain (analyze, verbose, costs off, timing off, summary off)
+select to_tsvector(text), to_tsvector('testing')
+from words where to_tsvector(text) != to_tsvector('testing');
+
+-- The default text search settings for data nodes should be the same as on the
+-- access node, so evaluating to_tsvector on data nodes should work.
+reset default_text_search_config;
+set timescaledb.pushdown_functions = 'stable';
+
+select to_tsvector(text), to_tsvector('testing')
+from words where to_tsvector(text) != to_tsvector('testing');
+
+explain (analyze, verbose, costs off, timing off, summary off)
+select to_tsvector(text), to_tsvector('testing')
+from words where to_tsvector(text) != to_tsvector('testing');
+
+-- A more complex test case -- add a volatile function that is not pushed down,
+-- to highlight the difference.
+select to_tsvector(text), to_tsvector('testing')
+from words where to_tsvector(text) != to_tsvector('testing')
+	and to_tsvector(text) != to_tsvector('testing' || random()::text);
+	
+explain (analyze, verbose, costs off, timing off, summary off)
+select to_tsvector(text), to_tsvector('testing')
+from words where to_tsvector(text) != to_tsvector('testing')
+	and to_tsvector(text) != to_tsvector('testing' || random()::text);
+
+-- Time for the weirdly broken stuff. This configuration will evaluate the
+-- to_tsvector('testing') on the access node, giving 'testing':1,
+-- and to_tsvector(text) on the data nodes, giving 'test':1. So this row will
+-- be returned from the data node. The access node won't check the filter again.
+-- To fix, we should start shipping the changed settings to the data nodes.
+-- For the time being, either configure the data nodes manually or limit the
+-- pushdown of functions to the whitelist (timezone stuff).
+set default_text_search_config = 'simple';
+set timescaledb.pushdown_functions = 'stable';
```

Comment:
I would suggest simple check to set this guc to something arbitrary, to make sure the value is properly validated by postgres

Reply:
Not sure it would add anything -- it's a enum GUC, int-valued, there's no way around validation.